client/core/core.go

```diff
@@ -995,34 +975,52 @@ func (c *Core) Network() dex.Network {
 	return c.net
 }
 
-// Exchanges returns a map of Exchange keyed by host, including a list of markets
-// and their orders.
+// Exchanges creates a map of *Exchange keyed by host, including markets and
+// orders.
 func (c *Core) Exchanges() map[string]*Exchange {
+	return c.exchangeMap()
+}
+
+// dexConnections creates a slice of the *dexConnection in c.conns.
+func (c *Core) dexConnections() []*dexConnection {
 	c.connMtx.RLock()
 	defer c.connMtx.RUnlock()
+	conns := make([]*dexConnection, 0, len(c.conns))
+	for _, conn := range c.conns {
+		conns = append(conns, conn)
+	}
+	return conns
```

Comment:
This was a pattern that @chappjc used in #921, and I wanted to use as well, and then realized that it really is better just about everywhere we were doing `range c.conns`. The same reasoning also applied to `c.wallets`, so I made the `xcWallets` method as well. Sorry for the scope expansion.